query/src/main/java/io/keen/client/java/KeenQueryRequest.java

```diff
@@ -0,0 +1,27 @@
+package io.keen.client.java;
+
+import io.keen.client.java.exceptions.KeenQueryClientException;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * Interface to be implemented by a query request
+ *
+ * @author baumatron
+ */
+abstract class KeenQueryRequest {
```

Comment:
Yeah, again we should do a follow-up push to limit the API surface of the SDK in general. I'd be all for that.